### ISSUE TITLE
Anchors/Aliases cannot be referenced for keys

### ISSUE THREAD

**Issue by Raffaello** [jbeder/yaml-cpp#908]
Defininng an alias in a Key won't produce the right output that require an extra spacing after column (`*a : value)`

see below for further detailed info.

----

*Detached from PR #903, original content:*

This kind of yaml file not sure is possible to emit:

```yml
Aliases:
 - {&i i: &v0 0}
 - {&i i: &v1 1}
NodeA:
  k: [{*i : *v0}, {*i : *v1}]
NodeB:
  k: [*i : *v0, *i : *v1]
```

should be a valid yaml, using to validate: https://yamlvalidator.com/

the actual test that is failing will be:
```c
TEST_F(EmitterTest, ComplexFlowSeqEmbeddingAMapUsingAliases4) {
  out << BeginMap;

  out << Key << "Aliases" << Value << BeginSeq;
  out << BeginMap << Anchor("i") << Key << "i" << Anchor("v0") << Value << 0
      << EndMap;
  out << BeginMap << Key << Anchor("i") << "i" << Value << Anchor("v1") << 1
      << EndMap;
  out << EndSeq;

  out << Key << "NodeA" << Value << BeginMap;
  out << Key << "k" << Value << Flow << BeginSeq;
  out << BeginMap << Key << Alias("i") << Value << Alias("v0") << EndMap;
  out << BeginMap << Key << Alias("i") << Value << Alias("v1") << EndMap;
  out << EndSeq << EndMap;

  out << Key << "NodeB" << Value << BeginMap;
  out << Key << "k" << Value << Flow << BeginSeq;
  out << Key << Alias("i") << Value << Alias("v0");
  out << Key << Alias("i") << Value << Alias("v1");
  out << EndSeq << EndMap;

  out << EndMap;

  ExpectEmit(R"(Aliases:
 - {&i i: &v0 0}
 - {&i i: &v1 1}
NodeA:
  k: [{*i : *v0}, {*i : *v1}]
NodeB:
  k: [*i : *v0, *i : *v1])");
}
```

_This test is going to fail also with the actual master code_, so should be considered in a different issue/PR to avoid scope creep in this one. Of course the analog sibling test checking the newline semantic will fail too.

Let me know your thoughts, i can commit anyway including this pair of tests and the CI should fail too.

p.s.
this test is the one asked for ` both key and value within each map`
[maybe consider a better naming for the review too, at this moment i am not in a really poetic mind form :)]

----

the current code emits:
````
[ RUN      ] EmitterTest.ComplexFlowSeqEmbeddingAMapUsingAliases4
..\..\..\test\integration\emitter_test.cpp(30): error: Expected equality of these values:
  expected
    Which is: "Aliases:\n - {&i i: &v0 0}\n - {&i i: &v1 1}\nNodeA:\n  k: [{*i : *v0}, {*i : *v1}]\nNodeB:\n  k: [*i : *v0, *i : *v1]"
  out.c_str()
    Which is: "Aliases:\n  - &i i: &v0 0\n  - &i i: &v1 1\nNodeA:\n  k: [{*i: *v0}, {*i: *v1}]\nNodeB:\n  k: [*i, *v0, *i, *v1]"
With diff:
@@ -1,7 +1,7 @@
 Aliases:
- - {&i i: &v0 0}
- - {&i i: &v1 1}
+  - &i i: &v0 0
+  - &i i: &v1 1
 NodeA:
-  k: [{*i : *v0}, {*i : *v1}]
+  k: [{*i: *v0}, {*i: *v1}]
 NodeB:
-  k: [*i : *v0, *i : *v1]
+  k: [*i, *v0, *i, *v1]

[  FAILED  ] EmitterTest.ComplexFlowSeqEmbeddingAMapUsingAliases4 (2 ms)
````

That should not neither a valid yaml (based on https://yamlvalidator.com/).

_Originally posted by @Raffaello in https://github.com/jbeder/yaml-cpp/pull/903_

**Comment by dota17** [comment]
Output analysis:
1
```yaml
Aliases:
  - &i i: &v0 0
  - &i i: &v1 1
```
Right.  Block is the default emitting format.

2
```yaml
NodeA:
  k: [{*i: *v0}, {*i: *v1}]
```
It's a Bug. The `:` should not follow the Aliase on the key directly. Because the `:` may be the part of aliase. Please refer to the #929.

3
```yaml
NodeB:
  k: [*i, *v0, *i, *v1]
```
Right. You should use the `out << BeginMap`&& `<< EndMap` to emit a map.

**Comment by dota17** [comment]
Fixed by #929. If still have other questions, please reopen it. @Raffaello
